resource_obmcs_core_instance.go

```diff
@@ -260,16 +260,19 @@ func (s *InstanceResourceCrud) Create() (e error) {
 	}
 
 	if rawVnic, ok := s.D.GetOk("create_vnic_details"); ok {
-		opts.CreateVnicOptions = SetCreateVnicOptions(rawVnic)
+		opts.CreateVnicOptions, e = SetCreateVnicOptions(rawVnic)
+	}
+
+	if e == nil {
```

Comment:
may look better to have an if  e != nil {return} right after the  SetCreateVnicOptions
but up to you